graphium/data/sampler.py

```diff
@@ -0,0 +1,27 @@
+import torch.utils.data as data_utils
+import numpy as np
+
+
+class CustomSampler(data_utils.Sampler):
+    def __init__(self, dataset, sampler_task_dict):
+        self.dataset = dataset
+        self.sampler_task_dict = sampler_task_dict
+        self.task_indices = {}
+        for i in range(len(dataset)):
+            task_name = dataset[i]["label"].name()
+            if task_name not in self.task_indices:
+                self.task_indices[task_name] = []
+            self.task_indices[task_name].append(i)
+
+    def __iter__(self):
```

Comment:
Is `__iter__` reinitialized every epoch or only at the beginning? Because we want the random sampling to change every epoch.

Reply:
I think it is reinitialized for each epoch. According to this discussion [here](https://stackoverflow.com/questions/75215951/pytorch-dataloader-across-epoch), the sequencial sampler and random sampler is applied for each epoch. And according to the pytorch [source code](https://pytorch.org/docs/stable/_modules/torch/utils/data/sampler.html#Sampler), the `CustomSampler` inherited from `data_utils.Sampler`, which has the `__iter__`, similar as the `SequentialSampler` and `RandomSampler`.